src/layer.js

```diff
@@ -148,6 +155,27 @@ export class MapLayer extends HTMLElement {
           this.opacity = newValue;
         }
       break;
+      case 'src':
```

Comment:
In the past, we've said that src takes precedence over content; here, if a value is provided for src where there wasn't one before and a MapMLLayer had / has already been created or exists, you create a shadow root. OK, so this means the layer is being transformed to a 'remote' one, and existing inline content is going to be ignored or removed, I believe.  Is that what you're shooting for here?  If so, I like it, but it also makes me wonder if we have thought out the ramifications.

Reply:
There are times when the src attribute is (re-)set by the MapMLLayer code, for example, when there is an alternate style selected by the user, or where there is a `<link rel=alternate projection="_a match for the current map's projection">`, so the client engages in "content selection" (for lack of a term).  There may be other use cases too, especially in the future for `<link rel=alternate media="a media query for text-optimized content (features)">`, that @AliyanH is looking into.  So it's already happening, and it makes sense to support it in the custom element `<layer->`.  Don't know if that rises to the level of useful, but fyi.

Reply:
Yes it is definitely useful! So for now we can keep it, and the codes should be modified to ensure the newly-added src attribute can overwrite the effect of original inline content, right? I think it is better to keep the inline content instead of just removing it as it might be useful later. WDYT?

Reply:
I agree.  The inline content is / can be a fallback.